CONTRIBUTING.md

```diff
@@ -1,78 +1,122 @@
- (e.g. `git checkout -b 12-new-feature`)
- - Make your changes
- - When committing your changes, make sure to start the commit message with `#<issue-number>`
- (e.g. `git commit -m '#12 - New Feature added'`)
+   (e.g. `git clone git@github.com:<YOUR_GITHUB_USER>/flame.git`).
+
+### Environment Setup
+Flame uses [Melos] to manage the project and dependencies.
+
+To install Melos, run the following command from your terminal:
+
+```bash
+flutter pub global activate melos
+```
+
+Next, at the root of your locally cloned repository bootstrap the projects dependencies:
+
+```bash
+melos bootstrap
+```
+
+The bootstrap command locally links all dependencies within the project without having to
+provide manual [`dependency_overrides`][pubspec doc]. This allows all
+plugins, examples and tests to build from the local clone project.
+
+> You do not need to run `flutter pub get` once bootstrap has been completed.
+
+### Performing changes
+ - Create a new local branch from `main` (e.g. `git checkout -b my-new-feature`)
+ - Make your changes.
+ - When committing your changes, make sure that each commit message is clear
+ (e.g. `git commit -m 'Take in an optional Camera as a parameter to FlameGame'`).
  - Push your new branch to your own fork into the same remote branch
- (e.g. `git push origin 12-new-feature`)
- - On Github goto the
- [pull request page](https://guides.github.com/activities/forking/#making-a-pull-request) on your
- own fork and create a merge request to this repository
-
-# Development Guidelines
- - Documentation should be updated.
- - Examples should be updated.
- - Format the Dart code accordingly.
- - Note the
- [`analysis_options.yaml`](https://github.com/flame-engine/flame/blob/main/analysis_options.yaml)
- and write code as stated in this file
-
-# Test generating of `dartdoc`
- - Run the `./scripts/dartdoc.sh` script to generate the Dart documentation.
- - Output will be placed into `doc/api/`.
+ (e.g. `git push origin my-new-feature`)
+
+### Open a pull request
+Go to the [pull request page of Flame][PRs] and in the top
+of the page it will ask you if you want to open a pull request from your newly created branch.
+
+The title of the pull request should start with a [conventional commit] type.
+
+Examples of such types:
+ - `fix:` - patches a bug and is not a new feature.
+ - `feat:` - introduces a new feature.
+ - `docs:` - updates or adds documentation or examples.
+ - `test:` - updates or adds tests.
+ - `refactor:` - refactors code but doesn't introduce any changes or additions to the public API.
+
+If you introduce a **breaking change** the conventional commit type MUST end with an exclamation
+mark (e.g. `feat!: remove the position argument from PositionComponent`).
+
+Examples of PR titles:
+ - feat: Component.childrenFactory can be used to set up a global ComponentSet factory
+ - fix: avoid infinite loop in `FlameGame`
+ - docs: added a `JoystickComponent` example
+ - docs: improved the Mandarin README
```

Comment:
according to the git commit message standards I follow, commit messages should be on imperative. should we maybe follow the same for PR titles too? eg: `docs: add a Foo example`. the conventional commits page follows that too

Reply:
also I am noticing all our open PRs use upper case after the flag (eg `docs: Improve...`). no strong opinion either way but might make sense to just formalize what is already there